site/src/pages/[shortname]/types/[title].page.tsx

```diff
@@ -0,0 +1,170 @@
+import Head from "next/head";
+import { tw } from "twind";
+import { NextPage } from "next";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+import {
+  BlockProtocolAggregateEntityTypesFunction,
+  BlockProtocolUpdateEntityTypesFunction,
+} from "blockprotocol";
+import { Box, Container, Typography } from "@mui/material";
+
+import { EntityType } from "../../../lib/model/entityType.model";
+import { SchemaEditor } from "../../../components/entityTypes/SchemaEditor/SchemaEditor";
+import { apiClient } from "../../../lib/apiClient";
+import { useUser } from "../../../context/UserContext";
+import { Link } from "../../../components/Link";
+
+type EntityTypePageQueryParams = {
+  shortname?: string;
+  title?: string;
+};
+
+const EntityTypePage: NextPage = () => {
+  const router = useRouter();
+
+  const { user } = useUser();
+
+  const [entityType, setEntityType] = useState<EntityType | undefined>();
+
+  // shortname and title will be undefined until useRouter populates them
+  const { shortname, title } = router.query as EntityTypePageQueryParams;
+
+  const shortnameWithoutLeadingAt = shortname?.replace(/^@/, "");
+
+  useEffect(() => {
+    if (!shortnameWithoutLeadingAt || !title) {
+      return;
+    }
+    apiClient
+      .getEntityTypeByUserAndTitle({
+        shortname: shortnameWithoutLeadingAt,
+        title,
+      })
+      .then(({ data }) => setEntityType(data?.entityType))
+      .catch((err) => {
+        // eslint-disable-next-line no-console -- @todo handle 404s and show to user
+        console.error(err);
+      });
+  }, [shortnameWithoutLeadingAt, title, setEntityType]);
+
+  const aggregateEntityTypes: BlockProtocolAggregateEntityTypesFunction =
+    () => {
+      if (!shortnameWithoutLeadingAt) {
+        throw new Error(
+          "Aggregate entity types called before shortname available from query.",
+        );
+      }
+      return apiClient
+        .getUserEntityTypes({ shortname: shortnameWithoutLeadingAt })
+        .then(({ data }) => {
+          return {
+            results: data?.entityTypes.map((type) => type.schema) ?? [],
+            operation: {
+              pageNumber: 1,
+              itemsPerPage: 100,
+            },
+          };
+        });
+    };
+
+  const updateEntityTypes: BlockProtocolUpdateEntityTypesFunction = ([
+    { entityTypeId, schema },
+  ]) =>
+    apiClient
+      .updateEntityType({ schema: JSON.stringify(schema) }, entityTypeId)
+      .then(({ data }) => {
+        if (data) {
+          return [data.entityType.schema];
+        }
+        throw new Error("Could not update entity type");
+      });
+
+  if (!entityType) {
+    return null;
```

Comment:
```js
import Error from "next/error";
```

```suggestion
    return <Error statusCode={404} />;
```

Otherwise `/@hash/types/adsf` gives us a blank page. It’d be ideal to move fetching to `getServerSideProps` so that we can return `{ notFound: true }` but that’s not something we want to do now.

Reply:
Actually we should introduce the loading state here too. Otherwise, we’ll see 404 while the entity is loading.

Reply:
Done